### PR TITLE
Angular: migrate Eslint configuration to recommended flat config

### PR DESCRIPTION
**Pull request by qmonmert**

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java [hunk 1]
```diff
@@ -22,7 +22,7 @@ public class AngularModuleFactory {
   public JHipsterModule buildModule(JHipsterModuleProperties properties) {
     //@formatter:off
     return moduleBuilder(properties)
-      .preCommitActions(stagedFilesFilter("{src/**/,}*.ts"), preCommitCommands("eslint --fix", "prettier --write"))
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:25]: Why are you removing prettier here?

**Review by murdos — COMMENTED** [review]
I was precisely trying to migrate eslint configuration to flat config, while investigating why there was linting issue on CI in https://github.com/jhipster/jhipster-lite/pull/10849

File: src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java (hunk 1 above)
Comment (qmonmert) [reply src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:25]: you added this line but there is no prettier in package.json now

Comment (murdos) [reply src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:25]: The angular-core module depends on the prettier module, so prettier should be present on a generated angular application: https://github.com/jhipster/jhipster-lite/blob/49ce34e84c05f5a2567cf59858280d7b4965cd2c/src/main/java/tech/jhipster/lite/generator/client/angular/core/infrastructure/primary/AngularCoreModuleConfiguration.java#L22

Comment (qmonmert) [reply src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:25]: generate angularapp you will see

Comment (murdos) [reply src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:25]: However in the generated apps on CI, the prettier module was wrongly not applied.
#10859 should fix that

Comment (murdos) [reply src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:25]: > generate angularapp you will see

Try on https://lite.jhipster.tech/landscape, you will see :D

Comment (qmonmert) [reply src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:25]: > However in the generated apps on CI, the prettier module was wrongly not applied. #10859 should fix that

ok I update

File: src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java [hunk 2]
```diff
@@ -56,12 +56,15 @@ public JHipsterModule buildModule(JHipsterModuleProperties properties) {
         .addDevDependency(packageName("@angular/compiler-cli"), ANGULAR, ANGULAR_CORE_PACKAGE)
         .addDevDependency(packageName("@types/node"), COMMON)
         .addDevDependency(packageName("@types/jest"), COMMON)
+        .addDevDependency(packageName("angular-eslint"), ANGULAR)
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:59]: We might probably remove the `@angular-eslint/*`packages.
See the documentation:
image

Comment (qmonmert) [reply src/main/java/tech/jhipster/lite/generator/client/angular/core/domain/AngularModuleFactory.java:59]: ok I do that

**Review by murdos — APPROVED** [review]

**Comment by murdos** [comment]
@qmonmert : I changed the title of the PR to better reflect its content. 
I think the original issue you had while committing was caused by the absence of prettier in the generated angularapp.

**Comment by qmonmert** [comment]
@murdos 🤝

**Pull request merged**